### ISSUE TITLE
[WARN] Cannot find rule for <class 'torch.nn.modules.container.Sequential'>. Treat it as zero Macs and zero Params.

### ISSUE THREAD

**Issue by hitachinsk** [Lyken17/pytorch-OpCounter#107]
Hello, I'm not familiar with this excellent tool, I wanna implement a CNN model using pytorch, and this model may contain some `Sequential` objects, which include some submodels because I want to make my codes more readable. 

However, when I test the FLOPS and parameters of my model, I encounter such warning `[WARN] Cannot find rule for <class 'torch.nn.modules.container.Sequential'>. Treat it as zero Macs and zero Params.` does it make some difference to the final result? I'm looking forward to your reply, thank you very much.

**Comment by chenxlin222** [comment]
hello，have you resolved the problem?

**Comment by MaxxieTao** [comment]
I use 'from thopimport profile' and meet the same issue.  Then I try  'from thop.profile import profile' and solve the problem

**Comment by chenxlin222** [comment]
I am very glad to hear from you and Thank you very much ！




陈琳
邮箱：201920504@mail.sdu.edu.cn
签名由 网易邮箱大师 定制


On 09/28/2020 00:05, MaxxieTao wrote:


I use 'from thopimport profile' and meet the same issue. Then I try 'from thop.profile import profile' and solve the problem

**Comment by hitachinsk** [comment]
@chenxlin222 Sounds interesting, I will take a shot, thank you.

**Comment by Wblossom** [comment]
@hitachinsk  Hi, have you resolved the problem? I met the same problem.

**Comment by Jackyinuo** [comment]
@hitachinsk  I have met the same problem.   have you resolved the problem?

**Comment by sunruina2** [comment]
maybe same problem?
    flops, params = profile(xtr_net, inputs=(dense, fly_emb_lst),custom_ops={})
    print(flops / 1000, ' / ', params / 1000)


[WARN] Cannot find rule for <class '__main__.mio_dense_layer'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class 'torch.nn.modules.container.Sequential'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class '__main__.simple_dense_network'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class '__main__.XtrModule'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class 'torch.nn.modules.container.ModuleList'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class 'torch.nn.modules.activation.Sigmoid'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class '__main__.simple_lhuc_network'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class 'torch.nn.modules.activation.Softmax'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class '__main__.transformer_component'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class '__main__.mmoe_layer'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class '__main__.PositionalEmbedding'>. Treat it as zero Macs and zero Params.
[WARN] Cannot find rule for <class '__main__.SnackModuleInner'>. Treat it as zero Macs and zero Params.
0.0  /  0.0

**Comment by yuyijie1995** [comment]
Have you solved it?

**Comment by kaaarho** [comment]
I met the same prob. Have you solved it?

**Comment by zhangqizky** [comment]
same here, beg for solution.

**Comment by Cassieyy** [comment]
Beg for solution...

**Comment by paleomoon** [comment]
Same question...

**Comment by ClarissaDHL** [comment]
Hi, I met the same question. Have u sovled the problem?

**Comment by Leojc** [comment]
I think this warning doesn't affect the result. The submodules in the nn.Sequential are calculated. The warning info is a little misleading though.

**Comment by Jasonx001** [comment]
If the rule does not generate computation, or it is not a part of your source architecture, just treat it as zero Macs and zero Params.

**Comment by ironman1996** [comment]
> I think this warning doesn't affect the result. The submodules in the nn.Sequential are calculated. The warning info is a little misleading though.

Actually it affects. With th warning, my network's macs and paras equal 0.

**Comment by zyxjtu** [comment]
> Same question, found an explanation at [another tool](https://github.com/sovrasov/flops-counter.pytorch/issues/52), seems it's ok to a container.

Excuse me, the calculate result of another tool is right?  And you verify it's result? I'm looking forward to your reply, thank you very much.

**Comment by zhangqizky** [comment]
您的邮件已收到，我会尽快回复，谢谢。张琦

**Comment by paleomoon** [comment]
@zyxjtu I didn't use the another, I just state the similar issues. Use this tool, and ignore the warning, it's ok.

**Comment by Lyken17** [comment]
Hi there, 

Could you try upgrade to the latest version and see the warning still continues?

**Comment by bigdaronlee163** [comment]
I meet the same problem previously,for the FLOPs counter,I use  [https://github.com/1adrianb/pytorch-estimate-flops](flops tool) and get the correct answer.

**Comment by sunsq1024** [comment]
> Hi there,
> 
> Could you try upgrade to the latest version and see the warning still continues?
到现在仍然有警告

**Comment by zhangqizky** [comment]
您的邮件已收到，我会尽快回复，谢谢。张琦

**Comment by SebastianWang9** [comment]
I have same problems.Beg for solution...

**Comment by zhangqizky** [comment]
您的邮件已收到，我会尽快回复，谢谢。张琦

**Comment by SebastianWang9** [comment]
您好，我已收到您的邮件，谢谢！

**Comment by Cassieyy** [comment]
您的邮件已收到哦我会及时处理哒~
